cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java

```diff
@@ -273,33 +253,11 @@ private void tagsQuerying(HttpRequest request, HttpResponder responder, List<Str
     }
   }
 
-  private void contextQuerying(HttpRequest request, HttpResponder responder,
-                               String context, List<String> metrics, String groupBy) {
-    try {
-      List<String> groupByTags = parseGroupBy(groupBy);
-      MetricQueryResult queryResult = executeQuery(request, parseTagValuesAsMap(context),
-                                                   groupByTags, metrics);
-      responder.sendJson(HttpResponseStatus.OK, queryResult);
-    } catch (IllegalArgumentException e) {
-      LOG.warn("Invalid request", e);
-      responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
-    } catch (Exception e) {
-      LOG.error("Exception querying metrics ", e);
-      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Internal error while querying for metrics");
-    }
-  }
-
   private MetricQueryResult executeQuery(HttpRequest request, Map<String, String> sliceByTags,
-                                         List<String> groupByTags, List<String> metrics) {
-    try {
-      MetricQueryRequest queryRequest = new MetricQueryRequest(sliceByTags, metrics, groupByTags);
-      setTimeRangeInQueryRequest(queryRequest, new QueryStringDecoder(request.getUri()).getParameters());
-      return executeQuery(queryRequest);
-    } catch (IllegalArgumentException e) {
```

Comment:
we don't want to differentiate between invalid request and internal_server_error ?